bus/client.go

```diff
@@ -331,6 +331,29 @@ func (c *Client) ContractsForSlab(shards []object.Sector) (contracts []Contract,
 	panic("unimplemented")
 }
 
+// Setting returns the value for the setting with given key.
+func (c *Client) Setting(key string) (setting string, err error) {
+	err = c.c.GET(fmt.Sprintf("/setting/%s", key), &setting)
+	return
+}
+
+// Settings returns the keys of all settings in the store.
+func (c *Client) Settings() (settings []string, err error) {
+	err = c.c.GET("/settings", &settings)
+	return
+}
+
+// UpdateSetting will update or insert the setting for given key with the given value.
+func (c *Client) UpdateSetting(key, value string) error {
+	return c.c.POST(fmt.Sprintf("/setting/%s/%s", key, value), nil, nil)
+}
+
+// RedundancySettings returns the configured redundancy settings.
+func (c *Client) RedundancySettings() (rs RedundancySettings, err error) {
+	err = c.c.GET("/settings/redundancy", &rs)
+	return
+}
```

Comment:
So just to reiterate on what I meant above. e.g. this could be something like

```go
 func (c *Client) UpdateSetting(key, value interface{}) error {
        v, _ := json.Marshal(value)
        escapedValue := url.EscapeMethodThatDoesntComeToMine(string(v))
 	return c.c.POST(fmt.Sprintf("/setting/%s/%s", key, escapedValue), nil, nil)
 }

 func (c *Client) Setting() (resp interface{}, err error) {
 	err = c.c.GET(fmt.Sprintf("/setting/%s",key),  rs)
 	return
 }

 func (c *Client) UpdateRedundancySetting(key, rs RedundancySetting) error {
 	return c.UpdateSetting("redundancy", rs)
 }

 func (c *Client) RedundancySettings() (rs RedundancySettings, err error) {
 	return c.Setting("redundancy", &rs)
 }
```

Instead of having a separate endpoint for redundancy settings.

Reply:
I thought it was nice to have every separate field in the settings table, so you can update parts of some settings object without having to know the entire settings object. I updated it though because I think it is better UX to do it this way 👍